src/platform/user/profile/utilities/index.js

```diff
@@ -63,19 +65,36 @@ export function mapRawUserDataToState(json) {
   };
 }
 
+// Flag to indicate an active session for initial page loads.
+// It's distinct from the currentlyLoggedIn state, which
+// serves as confirmation that the user is logged in and
+// as a trigger to properly update any components that subscribe to it.
+export const hasSession = () => localStorage.getItem('hasSession');
```

Comment:
I wonder if sometime down the road this value could be part of the Redux store in `initialState`, and used in conjunction with `isLoggedIn` and maybe `sessionExpired` or something.

Reply:
We need the flag in `localStorage` in order to to know at the beginning of a page load whether we should be making an API request at all (the initial fetch user). If we only have it in the Redux store, we would need it to be set to `true` in reaction to a successful API request, but we wouldn't know whether to make that request in the first place without some flag already being set.